src/Microsoft.VisualStudio.ProjectSystem.VisualBasic.VS/ProjectSystem/VS/Automation/VSImports.cs

```diff
@@ -0,0 +1,163 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.ComponentModel.Composition;
+using System.Linq;
+using EnvDTE;
+using Microsoft.VisualStudio.ProjectSystem.VS.ConnectionPoint;
+using VSLangProj;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Automation
+{
+    [Export(typeof(Imports))]
+    [Export(typeof(ImportsEvents))]
+    [AppliesTo(ProjectCapability.VisualBasic)]
+    internal class VSImports : ConnectionPointContainer,
```

Comment:
It's also really, really expensive to say walk over each import here - as a basic for-loop will read all the elements twice per loop (once for Count and once for the indexer). I really want to know if this the way CPS does this - seems like there's a bunch of consistency and performance issues with this class.